src/coreclr/nativeaot/Runtime/inc/CommonTypes.h

```diff
@@ -6,9 +6,16 @@
 
 #include <cstddef>
 #include <cstdint>
-#include <cstdlib>
+#include <stdlib.h>
+#include <stdio.h>
 #include <new>
 
+#ifdef TARGET_WINDOWS
+#define PURE_VIRTUAL = 0;
+#else
+#define PURE_VIRTUAL { assert(!"pure virtual function called"); while(true); }
```

Comment:
```suggestion
// Implement pure virtual on NativeAOT Unix (for -p:LinkStandardCPlusPlusLibrary=false the default),
// to avoid linker requiring __cxa_pure_virtual.
#ifdef TARGET_WINDOWS
#define PURE_VIRTUAL = 0;
#else
// `while(true);` is to satisfy the missing `return` statement. It will be optimized away by the compiler.
#define PURE_VIRTUAL { assert(!"pure virtual function called"); while(true); }
```